pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -3941,11 +3941,6 @@ func TestReconcileWithWhenExpressionsScopedToTask(t *testing.T) {
 	}, {
 		// was attempted, but has missing results references
 		Name: "e-task",
-		WhenExpressions: v1beta1.WhenExpressions{{
-			Input:    "$(tasks.a-task.results.aResult)",
-			Operator: "in",
-			Values:   []string{"aResultValue"},
-		}},
```

Comment:
Hey @jerop I think this was a nice feature to have it, `e-task` was skipped because the result reference `$(tasks.a-task.results.aResult)` is missing. After we add a reason to the skippedTask like you mentioned, it will be very obvious that we were not able to resolve a result reference. WDYT?

Reply:
that's right, people could deduce from the unresolved reference that that was the reason for skipping

Reply:
yea, maybe it makes sense to wait until we add the reason field before we make this cleanup